builder/vsphere/driver/vm.go

```diff
@@ -1096,29 +1101,39 @@ func findNetwork(network string, host string, d *VCenterDriver) (object.NetworkR
 	return nil, fmt.Errorf("error finding network; 'host' and 'network' not specified. at least one of the two must be specified")
 }
 
-func newVirtualPCIPassthroughAllowedDevice(devices []PCIPassthroughAllowedDevice) types.VirtualPCIPassthrough {
+func newVirtualPCIPassthroughAllowedDevice(devices []PCIPassthroughAllowedDevice) (types.VirtualPCIPassthrough, error) {
 	allowedDevices := make([]types.VirtualPCIPassthroughAllowedDevice, len(devices))
 	for i, device := range devices {
 		deviceId, err := strconv.ParseInt(device.DeviceId, 16, 32)
 		if err != nil {
-			// handle error, for example:
-			log.Printf("Error parsing DeviceId: %v\n", err)
-			continue
+			return types.VirtualPCIPassthrough{}, fmt.Errorf("error parsing DeviceId %s: %v", device.DeviceId, err)
 		}
+		if deviceId > math.MaxInt32 || deviceId < math.MinInt32 {
```

Comment:
Given the documentation of `ParseInt`, I would expect it to error if the returned value cannot fit on 32 bits, so this check seems unnecessary; I understand this silences what was reported, but IMHO this is basically guaranteed not to happen if we handle the error returned by `ParseInt` above

Reply:
Note: can confirm, this errors in case it is out-of-bound: https://play.golang.com/p/nn3VDOewJM4

Reply:
Updated in https://github.com/hashicorp/packer-plugin-vsphere/commit/4acaf070794ee50fed70c457f06a704cd6c98122. Given that `strconv.ParseInt` will indeed return an error if the value cannot fit within the specified bit size, the additional range check is redundant. We can rely on the error handling of `strconv.ParseInt` to catch out-of-bound values.